components/script/dom/promise.rs

```diff
@@ -490,6 +491,28 @@ impl Callback for WaitForAllRejectionHandler {
     }
 }
 
+/// The microtask for performing successSteps given « » in
+/// <https://webidl.spec.whatwg.org/#wait-for-all>.
+#[derive(JSTraceable, MallocSizeOf)]
+#[cfg_attr(crown, allow(crown::unrooted_must_root))]
```

Comment:
```suggestion
#[cfg_attr(crown, crown::unrooted_must_root_lint::must_root)]
```

Reply:
Oof that's the case, then we should change the Dom to DomRoot and remove the annotation. The compiler is pointing out that we're storing a value that must be rooted inside of values that do not get rooted, which is risky.